lib/rouge/lexers/powershell.rb

```diff
@@ -652,28 +97,47 @@ class Powershell < Shell
         rule %r/\s+/, Text
         rule %r/\$?"/, Str::Double, :double_quotes
         rule %r/\$'/, Str::Single, :ansi_string
-
         rule %r/'/, Str::Single, :single_quotes
-
         rule %r/\*/, Keyword
-
         rule %r/;/, Text
         rule %r/[^=\*\s{}()$"'`<]+/, Text
         rule %r/\d+(?= |\Z)/, Num
         rule %r/</, Text
         mixin :interp
       end
 
+      state :hashtable do 
+        rule %r/\w/, Name::Variable
+        rule %r/}/, Operator, :pop!
+        mixin :root
+      end
+
+      state :multiline do
+        rule %r/\.(#{MULTILINEKEYWORDS})/i, Keyword::Pseudo
+        rule %r/[\w,\d,\s,\.,\-,\,:\/,{,},<,>"*]/, Comment
+        rule %r/#>/, Comment, :pop!
+        mixin :root
+      end
+
+      state :heredoc do
+        rule %r/"@/, Operator, :pop!
+        rule %r/[^$\n]+/, Str::Heredoc
+        rule %r/[$]/, Str::Heredoc
+        mixin :data
+      end
+
       prepend :basic do
-        rule %r(<#[\s,\S]*?#>)m, Comment::Multiline
-        rule %r/#.*$/, Comment::Single
+        rule %r(#requires\s-version \d.\d*$),Comment::Preproc
+        rule %r(<#), Comment, :multiline
+        rule %r(@{), Operator, :hashtable
+        rule %r(@"), Operator, :heredoc
         rule %r/\b(#{OPERATORS})\s*\b/i, Operator
-        rule %r/\b(#{ATTRIBUTES})\s*\b/i, Name::Attribute
-        rule %r/\b(#{KEYWORDS})\s*\b/i, Keyword
+        rule %r/\b(#{ATTRIBUTES})\s*\b/i, Name::Builtin::Pseudo
+        rule %r/[a-z,A-Z,0-9]+?-[a-z,A-Z,0-9]*/, Generic::Strong
```

Comment:
Commas inside the character classes here are incorrect.

Can you really have `A+-B` as a…—Hmmm, what is a `Generic::Strong`?

Reply:
> Hmmm, what is a `Generic::Strong`?

@pyrmont https://github.com/rouge-ruby/rouge/wiki/List-of-tokens says it means
> Mark the token value as bold (for rst lexer)

Reply:
@ashmaroli That could have been phrased better. 

@aaroneg What does this text represent? Just text? A keyword? An identifier?

Reply:
Aaaahhhh.... I get it now. The Generic::Strong is to match ```Verb-Noun```  cmdlets in place of using built-ins (since technically they're not built-in commands). I assume you're using alphanumerics rather than \w so that you exclude underscores.

I think you want "+" instead of "*" for the noun -- would you match ```Get-``` as a valid cmdlet? i.e. ```rule %r/[a-zA-Z0-9]+?-[a-zA-Z0-9]+/, Generic::Strong```

Reply:
@pyrmont I'm just pulling from the available options on the Rackup legend. I suppose it could be Name::Function. In fact yes, that's what I'll change it to. 
I've adjusted the regex based on some quick research on rubular.com . 

And yes, it will now just match things that look like a cmdlet. And yes, tmk the only 'valid' characters for cmdlet verbs or nouns are [a-zA-Z\d] . tbh powershell will allow you to call a function whatever you want, within reason, it's just a style preference and helps us find obvious cmdlets. In the example output you'll see that it also marks `gitlab-runner` as a cmdlet even though it's not, but that's not a bug. ps command line allows you to run not-cmdlets inside it and if those commands happen to be named in a similar fashion to cmdlets, so be it.

Reply:
@aaroneg https://github.com/rouge-ruby/rouge/wiki/List-of-tokens